internal/pkg/term/selector/selector.go

```diff
@@ -427,6 +427,48 @@ func (s *Select) Environment(prompt, help, app string, additionalOpts ...string)
 	return selectedEnvName, nil
 }
 
+// Environments fetches all the environments in an app and prompts the user to select one OR MORE.
+// The List of options decreases as envs are chosen. Chosen envs displayed above with the finalMsg.
+func (s *Select) Environments(prompt, help, app string, finalMsg prompt.Option) ([]string, error) {
+	envs, err := s.retrieveEnvironments(app)
+	if err != nil {
+		return nil, fmt.Errorf("get environments for app %s from metadata store: %w", app, err)
+	}
+	if len(envs) == 0 {
+		log.Infof("Couldn't find any environments associated with app %s, try initializing one: %s\n",
+			color.HighlightUserInput(app),
+			color.HighlightCode("copilot env init"))
+		return nil, fmt.Errorf("no environments found in app %s", app)
+	}
+
+	envs = append(envs, "[No additional environments]")
+	var selectedEnvs []string
+	usedEnvs := make(map[string]bool)
+	// This for-loop ends with one item left to select because when Environments() is used for `pipeline init`,
+	// the "[No additional environments]" option will be the only choice remaining if the user selects all
+	// of their environments for their pipeline.
+	for i := 1; i < len(envs); i++ {
```

Comment:
Would it be slightly better?
```suggestion
	for {
	  ...
	  if selectedEnv == noMoreEnvs { 
	    break
	  }
	  selectedEnvs = append(selectedEnvs, selectedEnv)
	  // If only [No additional environments] is available
	  if len(selectedEnvs) == len(envs) - 1 {
	    break
	  }
	  usedEnvs[selectedEnv] = true
	}
```

Reply:
Ah, I like that! Done.